packages/dx-react-scheduler/src/plugins/day-view.tsx

```diff
@@ -20,90 +21,143 @@ import {
   VERTICAL_TYPE,
 } from '@devexpress/dx-scheduler-core';
 
+import { VerticalViewProps, ViewState } from '../types';
+
 const TYPE = 'day';
 const startViewDateBaseComputed = ({
   viewCellsData,
 }) => startViewDateCore(viewCellsData);
 const endViewDateBaseComputed = ({
   viewCellsData,
 }) => endViewDateCore(viewCellsData);
+const CellPlaceholder = params => <TemplatePlaceholder name="cell" params={params} />;
+const AppointmentPlaceholder = params => <TemplatePlaceholder name="appointment" params={params} />;
+const TimeTablePlaceholder = () => <TemplatePlaceholder name="main" />;
+const DayScaleEmptyCellPlaceholder = () => <TemplatePlaceholder name="dayScaleEmptyCell" />;
+const DayScalePlaceholder = () => <TemplatePlaceholder name="navbar" />;
+const SidebarPlaceholder = () => <TemplatePlaceholder name="sidebar" />;
 
-export class DayView extends React.PureComponent {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      timeTableRef: null,
-    };
+class DayViewBase extends React.PureComponent<VerticalViewProps, ViewState> {
+  timeTable: { current: HTMLElement | null } = { current: null };
```

Comment:
The type of `timeTable` is actually a RefObject. It's `current` property is mutable in React but readonly in definitions. We can fix it introducing a `WritableRefObject` type and use it in other views too.
```
type Mutable<T> = { -readonly [P in keyof T]: T[P] }; // to dx-core
type WritableRefObject<T> = Mutable<React.RefObject<T>> // to dx-react-core

timeTable: WritableRefObject<HTMLElement> = React.createRef();